htmresearch/algorithms/column_pooler.py

```diff
@@ -273,25 +282,52 @@ def _computeInferenceMode(self, feedforwardInput, lateralInputs):
         lateralInput, self.connectedPermanenceDistal)
       numActiveSegmentsByCell[overlaps >= self.activationThresholdDistal] += 1
 
-    # Activate some of the feedforward supported cells
-    minNumActiveCells = self.sdrSize / 2
-    chosenCells = self._chooseCells(feedforwardSupportedCells,
-                                    minNumActiveCells, numActiveSegmentsByCell)
+    chosenCells = []
+    minNumActiveCells =  30 #self.sdrSize / 2
 
-    # If necessary, activate some of the previously active cells
-    if len(chosenCells) < minNumActiveCells:
-      remaining = numpy.setdiff1d(prevActiveCells, feedforwardSupportedCells)
-      remaining = remaining[numActiveSegmentsByCell[remaining] > 0]
+    remainingFFSupportedCells = numpy.setdiff1d(feedforwardSupportedCells,
+                                                    chosenCells)
+    numActiveSegsForFFSuppCells=numActiveSegmentsByCell[remainingFFSupportedCells]
 
-      chosenCells = numpy.append(
-        chosenCells, self._chooseCells(remaining,
-                                       minNumActiveCells - len(chosenCells),
-                                       numActiveSegmentsByCell))
+    # First, activate the FF-supported cells that have the highest number of
+    # lateral active segments (this may be 0)
+    if len(remainingFFSupportedCells) == 0:
+        pass
+    else:
+      # This loop will select the FF-supported AND laterally-active cells, in
+      # order of descending lateral activation, until we exceed the
+      # minNumActiveCells quorum - but will exclude cells with 0 lateral
+      # active segments!
+      ttop = numpy.max(numActiveSegsForFFSuppCells)
+      if ttop > 0:
+        while True:
+          chosenCells = numpy.unique(numpy.append(chosenCells,
+                    remainingFFSupportedCells[numActiveSegsForFFSuppCells >
+                    self.distalSegmentInhibitionFactor * ttop]))
+          ttop -= 1
+          if ttop <= 0 or len(chosenCells) > minNumActiveCells:
+              break
```

Comment:
You could remove lines 302, 303, 308, 309, and just do 

~~~python
while ttop > 0 and len(chosenCells) > minNumActiveCells:
  ...
~~~